src/models/ingestion/ingestionHandler.ts

```diff
@@ -0,0 +1,212 @@
+import path from 'path';
+import fs from 'fs';
+import { StatusCodes } from 'http-status-codes';
+import { inject, injectable } from 'tsyringe';
+import { IJobResponse, ITaskResponse, IUpdateTaskBody, TaskHandler as QueueClient } from '@map-colonies/mc-priority-queue';
+import { Logger } from '@map-colonies/js-logger';
+import { MetricsCollector, ShapefileChunkReader, ProcessingState, ChunkProcessor, ShapefileChunk, StateManager } from '@map-colonies/mc-utils';
+
+import {
+  PartFeatureProperties,
+  PolygonPartsFeatureCollection,
+  PolygonPartsPayload,
+  rasterProductTypeSchema,
+  SHAPEFILE_EXTENSIONS_LIST,
+} from '@map-colonies/raster-shared';
+import { ZodError } from 'zod';
+import { FeatureResolutions, IConfig, IJobHandler, IngestionJobParams, ValidationsTaskParameters } from '../../common/interfaces';
+import { PolygonPartsManagerClient } from '../../clients/polygonPartsManagerClient';
+import { SERVICES } from '../../common/constants';
+import { ShpFeatureProperties, shpFeatureSchema } from '../../schemas/shpFile.schema';
+import { ShapefileMetrics } from '../../common/otel/metrics/shapeFileMetrics';
+import { ShapefileNotFoundError } from '../../common/errors';
+import { calculateResMeterFromDegree } from '../../utils/utils';
+
+@injectable()
+export class IngestionJobHandler implements IJobHandler<IngestionJobParams, ValidationsTaskParameters> {
+  private readonly maxVerticesPerChunk: number;
+  private readonly ingestionSourcesDirPath: string;
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.QUEUE_CLIENT) private readonly queueClient: QueueClient,
+    @inject(PolygonPartsManagerClient) private readonly polygonPartsManager: PolygonPartsManagerClient,
+    @inject(ShapefileMetrics) private readonly shapeFileMetrics: ShapefileMetrics,
+    @inject(SERVICES.CONFIG) private readonly config: IConfig
+  ) {
+    this.maxVerticesPerChunk = config.get<number>('jobDefinitions.tasks.validations.chunkMaxVertices');
+    this.ingestionSourcesDirPath = this.config.get<string>('ingestionSourcesDirPath');
+  }
+
+  public async processJob(job: IJobResponse<IngestionJobParams, unknown>, task: ITaskResponse<ValidationsTaskParameters>): Promise<void> {
```

Comment:
job param can now have a type of:
public async processJob(job: IJobResponse<IngestionJobParams, ***ValidationsTaskParameters***>, ...)